addons/rook/1.6.11/install.sh

```diff
@@ -0,0 +1,412 @@
+
+function rook_pre_init() {
+    local current_version='' current_version_minor='' current_version_patch=''
+    local next_version_minor='' next_version_patch=''
+
+    export SKIP_ROOK_INSTALL
+
+    current_version="$(rook_version)"
+    semverParse "${current_version}"
+    current_version_major="${major}"
+    current_version_minor="${minor}"
+    current_version_patch="${patch}"
+
+    semverParse "${ROOK_VERSION}"
+    next_version_minor="${minor}"
+    next_version_patch="${patch}"
+
+    if [ -n "${current_version}" ]; then
+        if [ "${current_version_minor}" != "${next_version_minor}" ]; then
+            if [ "${current_version_minor}" -gt "${next_version_minor}" ]; then
+                echo "Rook ${current_version} is already installed, will not downgrade to ${ROOK_VERSION}"
+                SKIP_ROOK_INSTALL=1
+            # upgrades from version 1.0.4 unsupported
+            elif [ "${current_version_major}" = "1" ] && [ "${current_version_minor}" = "0" ]; then
+                echo "Rook ${current_version} is already installed, will not upgrade to ${ROOK_VERSION}"
+                SKIP_ROOK_INSTALL=1
+                if [ "$KUBERNETES_TARGET_VERSION_MINOR" -ge 20 ]; then
+                    KUBERNETES_UPGRADE="0"
+                    KUBERNETES_VERSION=$(kubectl version --short | grep -i server | awk '{ print $3 }' | sed 's/^v*//')
+                    parse_kubernetes_target_version
+                    # There's no guarantee the packages from this version of Kubernetes are still available
+                    SKIP_KUBERNETES_HOST=1
+                fi
+            fi
+        elif [ "${current_version_patch}" -gt "${next_version_patch}" ]; then
+            echo "Rook ${current_version} is already installed, will not downgrade to ${ROOK_VERSION}"
+            SKIP_ROOK_INSTALL=1
+        fi
+    fi
+
+    if [ "${ROOK_BYPASS_UPGRADE_WARNING}" != "1" ]; then
+        if [ -z "${SKIP_ROOK_INSTALL}" ] && [ -n "${current_version}" ] && [ "${current_version}" != "${ROOK_VERSION}" ]; then
+            logWarn "WARNING: This installer will upgrade Rook to version ${ROOK_VERSION}."
```

Comment:
can we list the current rook version here?

IMO from 1.4 to 1.6 is more risky than from 1.5, and I'd like to expose that

Reply:
@laverya @divolgin is this request blocking merge